src/main/kotlin/org/jitsi/nlj/transform/node/incoming/IncomingStatisticsTracker.kt

```diff
@@ -37,42 +33,30 @@ import java.util.concurrent.ConcurrentHashMap
 /**
  * Track various statistics about received RTP streams to be used in SR/RR report blocks
  */
-class IncomingStatisticsTracker : ObserverNode("Incoming statistics tracker") {
+class IncomingStatisticsTracker(
+    private val streamInformationStore: StreamInformationStore
+) : ObserverNode("Incoming statistics tracker") {
     private val ssrcStats: MutableMap<Long, IncomingSsrcStats> = ConcurrentHashMap()
-    private val payloadTypes: MutableMap<Byte, PayloadType> = ConcurrentHashMap()
 
     override fun observe(packetInfo: PacketInfo) {
         val rtpPacket = packetInfo.packetAs<RtpPacket>()
-        payloadTypes[rtpPacket.payloadType.toByte()]?.let {
-            val stats = ssrcStats.computeIfAbsent(rtpPacket.ssrc) {
-                IncomingSsrcStats(rtpPacket.ssrc, rtpPacket.sequenceNumber)
-            }
-            val packetSentTimestamp = convertRtpTimestampToMs(rtpPacket.timestamp.toUInt(), it.clockRate)
-            stats.packetReceived(rtpPacket, packetSentTimestamp, packetInfo.receivedTime)
-        }
-    }
-
-    override fun handleEvent(event: Event) {
-        when (event) {
-            is RtpPayloadTypeAddedEvent -> {
-                // We don't want to track jitter, etc. for RTX streams
-                if (event.payloadType is RtxPayloadType) {
-                    logger.cdebug { "Ignoring RTX format: ${event.payloadType}" }
-                } else {
-                    payloadTypes[event.payloadType.pt] = event.payloadType
+        streamInformationStore.rtpPayloadTypes[rtpPacket.payloadType.toByte()]?.let {
+            // We don't want to track jitter, etc. for RTX streams
+            if (it !is RtxPayloadType) {
+                val stats = ssrcStats.computeIfAbsent(rtpPacket.ssrc) {
+                    IncomingSsrcStats(rtpPacket.ssrc, rtpPacket.sequenceNumber)
                 }
+                val packetSentTimestamp = convertRtpTimestampToMs(rtpPacket.timestamp.toUInt(), it.clockRate)
+                stats.packetReceived(rtpPacket, packetSentTimestamp, packetInfo.receivedTime)
             }
-            is RtpPayloadTypeClearEvent -> payloadTypes.clear()
         }
-        super.handleEvent(event)
     }
 
     override fun getNodeStats(): NodeStatsBlock {
         return super.getNodeStats().apply {
             val stats = getSnapshot()
             stats.ssrcStats.forEach { (ssrc, streamStats) ->
-                addString("source_ssrc", ssrc.toString())
-                addString("stream_stats", streamStats.toString())
+                addString(ssrc.toString(), streamStats.toString())
```

Comment:
i found that the old code here was actually overwriting itself, so we only tracked a single ssrc (the last one added)

Reply:
I think I meant to add a block for each SSRC, but this also works